src/app/(app)/home/entries/[entryId]/page.jsx

```diff
@@ -0,0 +1,34 @@
+import { getEntry } from "@/app/actions/entries"
+import Link from "next/link"
+import styles from './viewer.module.scss';
+
+
+export default async function Page({ params }) {
+  const entry = await getEntry(params.entryId)
+  return (
+    <section className={styles.viewerContainer}>
+      <div className="viewer-nav">
+        {/* <Link href="/home">back to list</Link> */}
+      </div>
+      
+      <section className={styles.viewerDetails}>
+          <div className={styles.createdTimeDate}>
+            <span className={styles.date}>{entry[0].created_at.toLocaleDateString('en-us', { weekday:"long", year:"numeric", month:"short", day:"numeric"})}</span>|
+            <span className={styles.date}>{entry[0].created_at.toLocaleTimeString('en-us', { timeStyle: 'short' })}</span>
+          </div>
+          <div className={styles.edit}>
+            <button className={styles.editButton}>Edit</button>
+            <span className={styles.editedText}>last edited {entry[0].updated_at.toLocaleDateString('en-us', { month:"short", day:"numeric"})}</span>
+          </div>
+      </section>
+
+      <section className={styles.viewerHeader}>
+        <h1 className={styles.title}>{entry[0].title}</h1>
+      </section>
+
+      <section className={styles.body}>
+        <p>{entry[0].body}</p>
+      </section>
+    </section>
+  )
+}
```

Comment:
Something I would challenge you to try is use entry title/creation date rather than id for pathing

Reply:
I totally agree! interestingly, it's not straightforward to do since at least with my current setup, whatever params I use to query the database (the id) has to be a part of the path too. Also the case with [generateStaticParams](https://nextjs.org/docs/app/api-reference/functions/generate-static-params).
A possible workaround would be for the entries to have their own `slug` property, so we can query them based on that (pretty much the example from the docs). But I'll try to look for other solutions first!

Reply:
Agreed, I think you are onto something with the `slug` prop. Might be a good opportunity to get your hands dirty with some SQL. I would be curious if you come up with anything else, but as I see it, a slug satisfies our technical requirements

Reply:
Cool! So I found out about triggers in postgres, which I think could work for generating the slug property whenever an entry is inserted. Do you have a suggested way for testing/experimenting with SQL code? I've been playing around a bit with Postico. But I don't want to accidentally do something destructive.

Reply:
Postgres clients recommendations is like recommending a wine. Its seasonal and up to taste. Personally I just use `psql` and connect directly through the terminal because I just dont have the palette for wine.

In regards to worrying about destructive actions, dont worry. You are not connect to prod so breaking your local and spending hours debugging has no rippling effects, plus its where all the best learning happens! If push comes to shove and you fully nuked your db you can always rerun `postgres::seed` and start from scratch. No harm. No foul.

I tried to make the tool bulletproof so if you run into issues just let me know and I an patch them, or, if you feel like taking a stab at it go for it